debezium-api/src/main/java/io/debezium/spi/converter/RelationalColumn.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.spi.converter;
+
+import java.sql.Types;
+import java.util.Optional;
+
+import io.debezium.common.annotation.Incubating;
+
+/**
+ * A definition of a converted relational column.
+ *
+ * @author Randall Hauch
+
+ */
+@Incubating
+public interface RelationalColumn extends ConvertedField {
+
+    /**
+     * Get the {@link Types JDBC type} for this column
+     *
+     * @return the type constant
+     */
+    int jdbcType();
+
+    /**
+     * Get the database native type for this column
+     *
+     * @return a type constant for the specific database
+     */
+    int nativeType();
+
+    /**
+     * Get the database-specific name of the column's data type.
+     *
+     * @return the name of the type
+     */
+    String typeName();
+
+    /**
+     * Get the database-specific complete expression defining the column's data type, including dimensions, length, precision,
+     * character sets, constraints, etc.
+     *
+     * @return the complete type expression
+     */
+    String typeExpression();
+
+    /**
+     * Get the maximum length of this column's values. For numeric columns, this represents the precision.
+     *
+     * @return the length of the column
+     */
+    int length();
+
+    /**
+     * Get the scale of the column.
+     *
+     * @return the scale if it applies to this type
+     */
+    Optional<Integer> scale();
+
+    /**
+     * Determine whether this column is optional.
+     *
+     * @return {@code true} if it is optional, or {@code false} otherwise
+     */
+    boolean isOptional();
+
+    /**
+     * Get the default value of the column
+     *
+     * @return the default value
+     */
+    Object defaultValue();
```

Comment:
`Optional<Object>`?

Reply:
This is a little bit tricky as there is three-state value

 - no default value
 - default value set to null
 - defalt value

The problem is that it is a little bit cunterintuitive to use Optional to express it.

We could use logic

- null - no default value
- Optional.empty() - NULL default value
- Optional.of - default value

The problem is that the usual convention is that Optional methods does not return null. So that's why we opted for defaultValue/hasDefaultValue combo.

The most clean solution migt be to use `<Optional<Optional<Object>>` but that onle looks little bit ugly.

Reply:
Ah yes, I remember that discussion now. The current design seems most reasonable in that light.